wgsl/index.bs

```diff
@@ -3959,6 +3959,261 @@ TODO: deduplicate these tables
+</table>
+
+<table class='data'>
+  <caption>Compatibility with other named arguments</caption>
+  <thead>
+    <tr><td><td>`offset`<td>`bias` <td>`lod`  <td>`grad`
+  </thead>
+  <tr><td> `offset` <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `bias`   <td>&#10003;<td>        <td>        <td>
+  <tr><td> `lod`    <td>&#10003;<td>        <td>        <td>
+  <tr><td> `grad`   <td>&#10003;<td>        <td>        <td>
+</table>
+
+### Sample & Compare ### {#texture-sample-compare-functions}
+
+Sampling from a depth texture with a depth-comparison uses the `textureSampleCompare` function:
+
+```
+f32 textureSampleCompare(<depth_texture>, <sampler_comparison>, <coordinates>, f32 depth_reference [, offset: <offset>])
+```
+
+Where:
+
+* `<depth_texture>` is a [depth](#texture-depth) texture type.
+
+* `<sampler_comparison>` is a [sampler comparision](#sampler-type) type.
+
+* `<coordinates>` are the texture coordinates used for sampling.
+    The type for the texture coordinates depends on the `<texture>` type:
+
+    <table class='data'>
+      <thead>
+        <tr><td>`<texture>` type<td>`<coordinates>` type
+      </thead>
+      <tr><td> `texture_depth_2d`         <td> `vec2<f32>`
+      <tr><td> `texture_depth_2d_array`   <td> `vec3<f32>`
+      <tr><td> `texture_depth_cube`       <td> `vec3<f32>`
+      <tr><td> `texture_depth_cube_array` <td> `vec4<f32>`
+    </table>
+
+    Array textures (with `_array` suffix) use the last component of `<coordinates>`
+    to specify array index. This value is rounded to the nearest integer and is
+    clamped to the array range before sampling.
+
+
+Samples the depth value at `<coordinates>`, and compares the sampled value against `depth_reference`
+and returns an `f32` in the range `[0.0 .. 1.0]`.
+The comparision operator is defined by the `<sampler_comparison>`.
+
+#### Sample & Compare with texel offset #### {#texture-sample-compare-with-offset}
+
+Following the `depth_reference` parameter, an optional named parameter for specifying an integer texel offset may be specified:
+
+```
+<result> textureSampleCompare(<depth_texture>, <sampler_comparison>, <coordinates>, f32 depth_reference, offset: <offset>)
+```
+
+If specified, the integer texel offset `<offset>` is applied to the unnormalized texture coordinate before sampling.
+
+TODO(ben-clayton): Check if the addressing modes of the sampler (e.g. wrapping) are applied before this offset or after.
+
+`<offset>` values may range from `[-8 .. 7]`. Values outside of this range will be clamped to the nearest limit.
+
+The type of `<offset>` is dependent on the type of `<texture>` and is either a `vec2<i32>` or `vec3<i32>`:
+
+<table class='data'>
+  <thead>
+    <tr><td>`<texture>` type<td>`<coordinates>` type
+  </thead>
+  <tr><td> `texture_depth_2d`                 <td> `vec2<i32>`
+  <tr><td> `texture_depth_2d_array`           <td> `vec2<i32>`
+  <tr><td> `texture_depth_cube`               <td> `vec3<i32>`
```

Comment:
we can't have an offset into cubes, as indicated in https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-to-samplecmp

Reply:
Thanks, I thought I had removed this, but apparently only for `textureSample()`. Done.